bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,1044 @@
+		}
+	};
+	private FocusListener overlayFocusListener = new FocusListener() {
+
+		@Override
+		public void focusGained(FocusEvent e) {
+			// do nothing
+		}
+
+		@Override
+		public void focusLost(FocusEvent e) {
+			findReplaceLogic.activate(SearchOptions.GLOBAL);
+			searchInSelectionButton.setSelection(false);
+		}
+
+	};
+	private PaintListener widgetMovementListener = new PaintListener() {
+
+		@Override
+		public void paintControl(PaintEvent e) {
+			positionToPart();
+		}
+
+	};
+	private IPartListener partListener = new IPartListener() {
+		@Override
+		public void partActivated(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partDeactivated(IWorkbenchPart part) {
+			// Do nothing
+		}
+
+		@Override
+		public void partBroughtToTop(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partClosed(IWorkbenchPart part) {
+			close();
+		}
+
+		@Override
+		public void partOpened(IWorkbenchPart part) {
+			// Do nothing
+		}
+	};
+
+	private boolean isPartCurrentlyDisplayedInPartSash() {
+		IWorkbenchPage activePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+
+		// Check if the targetPart is currently displayed on the active page
+		boolean isPartDisplayed = false;
+
+		if (activePage != null) {
+			IWorkbenchPart activePart = activePage.getActivePart();
+			if (activePart != null && activePart == targetPart) {
+				isPartDisplayed = true;
+			}
+		}
+
+		return isPartDisplayed;
+	}
+
+	/**
+	 * Returns the dialog settings object used to share state between several
+	 * find/replace overlays.
+	 *
+	 * @return the dialog settings to be used
+	 */
+	private static IDialogSettings getDialogSettings() {
+		IDialogSettings settings = PlatformUI
+				.getDialogSettingsProvider(FrameworkUtil.getBundle(FindReplaceOverlay.class)).getDialogSettings();
+		return settings;
+	}
+
+	@Override
+	public void create() {
+		if (overlayOpen) {
+			return;
+		}
+		super.create();
+	}
+
+	@Override
+	public boolean close() {
+		if (!overlayOpen) {
+			return true;
+		}
+		findReplaceLogic.activate(SearchOptions.GLOBAL);
+		getDialogSettings().put(REPLACE_BAR_OPEN_DIALOG_SETTING, replaceBarOpen);
+		overlayOpen = false;
+		replaceBarOpen = false;
+		unbindListeners();
+		container.dispose();
+		return super.close();
+	}
+
+	@Override
+	public int open() {
+		int returnCode;
+		if (overlayOpen) {
+			returnCode = Window.CANCEL;
+		} else {
+			bindListeners();
+			returnCode = super.open();
+		}
+		overlayOpen = true;
+		applyOverlayColors(backgroundToUse, true);
+		initFindStringFromSelection();
+
+		boolean shouldOpenReplaceBar = getDialogSettings().getBoolean(REPLACE_BAR_OPEN_DIALOG_SETTING);
+		if ((shouldOpenReplaceBar || Objects.isNull(shouldOpenReplaceBar)) && Objects.nonNull(replaceToggle)) {
```

Comment:
A `boolean` value cannot be `null`. You need to define a proper default value (true) when calling `getBoolean()` to retrieve the property.

Reply:
The `IDialogSettings`-interface does not expose a method to retrieve a preference with a default-value.
Instead, I will change the type of the variable to `Boolean` s.t. a null-check makes sense.

Reply:
I'm being super nit picky, but is `Objects.isNull(shouldOpenReplaceBar)` really better than `shouldOpenReplaceBar == null`?

The documentation for the method it says:
```
     * @apiNote This method exists to be used as a
     * {@link java.util.function.Predicate}, {@code filter(Objects::isNull)}
```
So personally I'm surprised to see it used in a context where it's not being used as a lambda.

Reply:
@merks I have seen it used somewhere in Eclipse - but seeing the docs, you are absolutely right, "Object.isNull" should really not be used here.

Reply:
Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/0d41d1995b249586aeb224baff70f8c06c13c495